packages/core/src/ui/widgets/colorBar/EventListeners.ts

```diff
@@ -0,0 +1,269 @@
+enum EventListenerPhases {
+  None = 0,
+  Capture = 1,
+  Bubble = 2,
+}
+
+type ListenersMap = Map<EventListener, EventListenerPhases>;
+
+class TargetEventListeners {
```

Comment:
This seem to be a useful class. Take it out please and put it in the core utils. Also please add a doc on top of it so that we know how to make use of it. The naming I think can be `EventListenersManager` or even `MultiTargetEventListenerManager`